src/lib.rs

```diff
@@ -90,22 +124,53 @@ pub fn read_ase<T: std::io::Read>(mut ase: T) -> Result<(Vec<Group>, Vec<ColorBl
         match block_type {
             BlockType::GroupStart => {
                 let block = Group::parse(&block)?;
-                groups.push(block);
-
-                // read the group end block
-                ase.read_exact(&mut buf_u16)?;
-                if BlockType::try_from(u16::from_be_bytes(buf_u16))? != BlockType::GroupEnd {
-                    // group has no end, file is invalid
+                if group_hold != types::GroupHold::Empty {
                     return Err(ASEError::Invalid(error::ConformationError::GroupEnd));
                 }
+                // if the parsed block has any sub-blocks then it has already been built
+                // and only a group-end block may follow it. Otherwise we are free to
+                // add colors as they appear until a group-end block is encountered.
+                group_hold = if block.blocks.is_empty() {
+                    types::GroupHold::HoldingBuilding
+                } else {
+                    blocks_to_read += 1;
+                    types::GroupHold::HoldingBuilt
+                };
```

Comment:
```suggestion
                group_hold = if block.blocks.is_empty() {
                    GroupHold::HoldingBuilding
                } else {
                    blocks_to_read += 1;
                    GroupHold::HoldingBuilt
                };
```

`types::GroupHold` is already imported. Please also adjust the other usages.